### PR TITLE
Integrate indexer `feat/altered-accounts`

### PR DESCRIPTION
**Pull request by miiu96**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- Integrate the latest indexer 
- Added a new driver that will send data over WS
  
## Proposed Changes
- Integrated a new indexer that should work without `TransactionsFeeHandler`

## Testing procedure
- Check the transaction fee and the gas used is set correctly in Elasticsearch

### CODE REVIEW

File: integrationTests/vm/testIndexer.go [hunk 1]
```diff
@@ -1,368 +0,0 @@
...
-	"github.com/ElrondNetwork/elrond-go-core/hashing"
-	"github.com/ElrondNetwork/elrond-go-core/marshal"
-	"github.com/ElrondNetwork/elrond-go/integrationTests/mock"
-	"github.com/ElrondNetwork/elrond-go/outport"
-	"github.com/ElrondNetwork/elrond-go/process"
-	"github.com/ElrondNetwork/elrond-go/sharding"
-	"github.com/stretchr/testify/require"
-)
-
-type testIndexer struct {
```
Comment (iulianpascalau) [inline integrationTests/vm/testIndexer.go:39]: suggestion to keep this in a form or another, not necessarily to output data in an indexer 
Reason: with this we test the integration between the processor & outport driver

File: outport/factory/outportFactory.go [hunk 2]
```diff
@@ -4,16 +4,23 @@ import (
 	"time"
 
 	covalentFactory "github.com/ElrondNetwork/covalent-indexer-go/factory"
-	indexerFactory "github.com/ElrondNetwork/elastic-indexer-go/factory"
+	indexerFactory "github.com/ElrondNetwork/elastic-indexer-go/process/factory"
+	wsDriverFactory "github.com/ElrondNetwork/elrond-go-core/websocketOutportDriver/factory"
 	"github.com/ElrondNetwork/elrond-go/outport"
 )
 
+type WrappedOutportDriverWebSocketSenderFactoryArgs struct {
```
Comment (iulianpascalau) [inline outport/factory/outportFactory.go:12]: missing comment

File: outport/process/transactionsfee/transactionsFeeProcessor.go [hunk 3]
```diff
@@ -116,15 +121,20 @@ func (tep *transactionsFeeProcessor) prepareTxWithResultsBasedOnLogs(
 	}
 
 	for _, event := range txWithResults.log.GetLogEvents() {
-		identifier := string(event.GetIdentifier())
-		shouldIgnore := identifier != core.SignalErrorOperation && identifier != core.WriteLogIdentifier
-		if shouldIgnore || hasRefund {
-			continue
+		switch {
```
Comment (iulianpascalau) [inline outport/process/transactionsfee/transactionsFeeProcessor.go:124]: I would separate this in a new function so the switch can be eliminated by using early exits

File: go.mod [hunk 4]
```diff
@@ -6,13 +6,13 @@ require (
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.42-0.20220729115258-b9f2fb2f6568
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.42-0.20220729115131-85ecca868e90
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_4 v1.4.59-0.20220729115431-a6c93119bdda
-	github.com/ElrondNetwork/covalent-indexer-go v1.0.7-0.20220729091107-1f75a52be6e9
-	github.com/ElrondNetwork/elastic-indexer-go v1.2.35-partial-execution.0.20220729082923-f6dd4fb19165
-	github.com/ElrondNetwork/elrond-go-core v1.1.20-0.20220831091042-a8722ecd7fba
+	github.com/ElrondNetwork/covalent-indexer-go v1.0.7-0.20220922092743-7f3a26b4001a
```
Comment (iulianpascalau) [inline go.mod:9]: let's not forget about proper releases. Can be done in another PR

File: cmd/node/config/external.toml [hunk 5]
```diff
@@ -47,3 +47,10 @@
     RouteSendData = "/block"
     # Route used to acknowledge sent blocks
     RouteAcknowledgeData = "/acknowledge"
+
+[WebSocketsConnector]
+    # This flag shall only be used for observer nodes
+    Enabled = false
+    URL = "localhost:22111"
+    WithAcknowledge = true
+    MarshallerType = "json"
```
Comment (bogdan-rosianu) [inline cmd/node/config/external.toml:56]: document this config value. could be like
`# Currently, only "json" is supported. In the future, "gogo protobuf" could also be supported`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: outport/factory/outportFactory.go (hunk 2 above)
Comment (miiu96) [reply outport/factory/outportFactory.go:12]: added

File: outport/process/transactionsfee/transactionsFeeProcessor.go (hunk 3 above)
Comment (miiu96) [reply outport/process/transactionsfee/transactionsFeeProcessor.go:124]: refactored

File: go.mod (hunk 4 above)
Comment (miiu96) [reply go.mod:9]: will do

File: cmd/node/config/external.toml [hunk 6]
```diff
@@ -47,3 +47,10 @@
     RouteSendData = "/block"
     # Route used to acknowledge sent blocks
     RouteAcknowledgeData = "/acknowledge"
+
+[WebSocketsConnector]
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/config/external.toml:51]: done.

File: config/externalConfig.go [hunk 7]
```diff
@@ -5,6 +5,7 @@ type ExternalConfig struct {
 	ElasticSearchConnector ElasticSearchConfig
 	EventNotifierConnector EventNotifierConfig
 	CovalentConnector      CovalentConfig
+	WebSocketsConnector    WebSocketsDriverConfig
```
[the remark this answers is not in the document]
Comment (miiu96) [reply config/externalConfig.go:8]: done.

File: cmd/node/config/external.toml (hunk 5 above)
Comment (miiu96) [reply cmd/node/config/external.toml:56]: done.

File: factory/status/statusComponents.go [hunk 8]
```diff
@@ -271,6 +279,28 @@ func (scf *statusComponentsFactory) makeCovalentIndexerArgs() *covalentFactory.A
 	}
 }
 
+func (scf *statusComponentsFactory) makeWebSocketsDriverArgs() (wsDriverFactory.OutportDriverWebSocketSenderFactoryArgs, error) {
```
Comment (bogdan-rosianu) [inline factory/status/statusComponents.go:282]: WebSocket*

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: outport/factory/outportFactory.go [hunk 9]
```diff
@@ -110,3 +118,24 @@ func checkArguments(args *OutportFactoryArgs) error {
 
 	return nil
 }
+
+func createAndSubscribeWebSocketsDriver(
```
[the remark this answers is not in the document]
Comment (miiu96) [reply outport/factory/outportFactory.go:122]: done

File: factory/status/statusComponents.go (hunk 8 above)
Comment (miiu96) [reply factory/status/statusComponents.go:282]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**